var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -0,0 +1,140 @@
+from spack import *
+
+import os
+
+
+class Dftbplus(MakefilePackage):
+    """DFTB+ is an implementation of the
+    Density Functional based Tight Binding (DFTB) method,
+    containing many extensions to the original method."""
+
+    homepage = "https://www.dftbplus.org"
+    url      = "https://www.dftbplus.org/fileadmin/DFTBPLUS/public/dftbplus/19.1/dftbplus-19.1.tar.xz"
+
+    version('19.1', sha256='78f45ef0571c78cf732a5493d32830455a832fa05ebcad43098895e46ad8d220')
+
+    variant('mpi', default=True,
+            description="Build an MPI-paralelised version of the code.")
+
+    variant('gpu', default=False,
+            description="""Use the MAGMA library
+            for GPU accelerated computation""")
+
+    variant('elsi', default=False,
+            description="""Use the ELSI library for large scale systems.
+            Only has any effect if you build with '+mpi'""")
+
+    variant('sockets', default=False,
+            description="""Whether the socket library
+            (external control) should be linked""")
+
+    variant('arpack', default=False,
+            description="Use ARPACK for excited state DFTB functionality")
+
+    variant('transport', default=False,
+            description="""Whether transport via libNEGF should be included.
+            Only affects parallel build.
+            (serial version is built without libNEGF/transport)""")
+
+    variant('dftd3', default=False,
+            description="""Use DftD3 dispersion library
+            (if you need this dispersion model)""")
+
+    depends_on('lapack')
+    depends_on('blas')
+    depends_on('scalapack', when="+mpi")
+    depends_on('mpi', when="+mpi")
+    depends_on('elsi', when="+elsi")
+    depends_on('magma', when="+gpu")
+    depends_on('arpack-ng', when="+arpack")
+    depends_on('dftd3-lib@0.9.2', when="+dftd3")
+
+    def edit(self, spec, prefix):
+        """
+        First, change the ROOT variable, because, for some reason,
+        the Makefile and the spack install script run in different directories
+
+        Then, if using GCC, rename the file 'sys/make.x86_64-linux-gnu'
+        to make.arch.
+
+        After that, edit the make.arch to point to the dependencies
+
+        And the last thing we do here is to set the installdir
+        """
+        dircwd = os.getcwd()
+        makefile = FileFilter("makefile")
+        makefile.filter("ROOT := .*", "ROOT := {}".format(dircwd))
```

Comment:
```suggestion
        makefile.filter("ROOT := .*", "ROOT := {0}".format(dircwd))
```
Otherwise this won't work in Python 2.6